### ISSUE TITLE
Error response from daemon: client version 1.40 is too new

### ISSUE THREAD

**Issue by nikolaykuz** [testcontainers/testcontainers-go#61]
I am using latest Docker for Mac 2.0.0.3, Engine: 18.09.2.

```
=== RUN   TestNginxLatestReturn
--- FAIL: TestNginxLatestReturn (0.01s)
    app_test.go:23: failed to create container: creating reaper failed: Error response from daemon: client version 1.40 is too new. Maximum supported API version is 1.39
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1739281]

goroutine 36 [running]:
testing.tRunner.func1(0xc000141900)
	/usr/local/Cellar/go/1.12/libexec/src/testing/testing.go:830 +0x388
panic(0x17fc540, 0x1fdf940)
	/usr/local/Cellar/go/1.12/libexec/src/runtime/panic.go:522 +0x1b5
```

**Comment by gianarb** [comment]
Hello! I was able to reproduce this issue, thank you! A possible workaround is to force the docker api version with the env var `DOCKER_API_VERSION`.
```
DOCKER_API_VERSION='1.39' go test ./...
ok      github.com/testcontainers/testcontainers-go     23.345s
?       github.com/testcontainers/testcontainers-go/wait        [no test files]
```

**Comment by bgabbasov-aurea** [comment]
I face same issue on windows using PowerShell. Setting DOCKER_API_VERSION variable does not help :(

**Comment by gianarb** [comment]
@bgabbasov-aurea what version of testcontainers are you using? This got fixed during the last release `v0.0.4` commit `0416f14` thanks

**Comment by adampatterson** [comment]
I ran the same command three times with three different results. 

`docker exec -it container_name bash`
Time Out

`docker exec -it container_name bash`
cannot connect to the Docker daemon. Is 'docker daemon' running on this host?: dial tcp 54.191.67.44:2376: connect: operation timed out

`docker exec -it container_name bash`
Worked as expected 🤷‍♂️
